enterprise/frontend/src/metabase-enterprise/sandboxes/components/GTAPModal.jsx

```diff
@@ -119,10 +123,21 @@ export default class GTAPModal extends React.Component {
     if (!gtap) {
       throw new Error("No GTAP");
     }
-    if (gtap.id != null) {
-      await GTAPApi.update(gtap);
-    } else {
-      await GTAPApi.create(gtap);
+    try {
+      if (gtap.id != null) {
+        await GTAPApi.update(gtap);
+      } else {
+        await GTAPApi.create(gtap);
+      }
+    } catch (error) {
+      console.error("Error saving GTAP", error);
+      const message = error
+        ? error.data
+          ? error.data.message || JSON.stringify(error.data)
+          : JSON.stringify(error)
+        : UNKNOWN_ERROR_MESSAGE;
+      this.setState({ error: message });
+      throw new Error(message);
```

Comment:
Why rethrow? To avoid closing the modal?

Reply:
yes